File: src/main/java/com/tkj/garagedoorz/health/DoorStatusHealthIndicator.java
package com.tkj.garagedoorz.health;

import java.util.List;

import org.springframework.boot.actuate.health.AbstractHealthIndicator;
import org.springframework.boot.actuate.health.Health.Builder;

import com.tkj.garagedoorz.domain.DoorStatus;
import com.tkj.garagedoorz.domain.HwController;

/**
 * HealthIndicator that reports garage door statuses.
 * @author Thomas G. Kenny Jr
 *
 */
public class DoorStatusHealthIndicator extends AbstractHealthIndicator {

	private final HwController hwController;
	
	public DoorStatusHealthIndicator(final HwController hwController) {
		this.hwController = hwController;
	}
	
	@Override
	protected void doHealthCheck(Builder builder) throws Exception {
		List<DoorStatus> doorStatuses = hwController.getGarageDoorStatuses();
		builder.up();
		builder.withDetail("doors", doorStatuses);
	}

}

File: src/main/java/com/tkj/garagedoorz/health/DoorStatusHealthIndicator.kt
package com.tkj.garagedoorz.health

import com.tkj.garagedoorz.domain.HwController
import org.springframework.boot.actuate.health.AbstractHealthIndicator
import org.springframework.boot.actuate.health.Health.Builder
import org.springframework.stereotype.Component

/**
 * HealthIndicator that reports garage door statuses.
 * @author Thomas G. Kenny Jr
 */
@Component
class DoorStatusHealthIndicator( private val hwController: HwController ) : AbstractHealthIndicator() {

    @Throws( Exception::class )
    override fun doHealthCheck( builder: Builder ) {

        val doorStatuses = hwController.garageDoorStatuses
        builder
                .up()
                .withDetail( "doors", doorStatuses )

    }

}
